File: 217-contains-duplicate/217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        
        ht = {}
        
        for num in nums:
            if num not in ht:
                ht[num] = 1
            else:
                ht[num]+=1
                
            if ht[num] > 1:
                return True
        
        return False
        

File: 217-contains-duplicate/217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> values;
        
        for(int& num : nums) {
            if(values.count(num)) {
                return true;
            } else {
                values.insert(num);
            }
        }
        
        return false;
        
    }
};